lib/smart_answer/money.rb

```diff
@@ -9,15 +9,9 @@ class Money
     attr_reader :value
 
     def initialize(raw_input)
-      if raw_input.is_a?(Numeric)
-        @value = BigDecimal.new(raw_input.to_s)
-      else
-        raw_input = raw_input.to_s.delete(',').gsub(/\s/, '')
-        if !self.class.valid?(raw_input)
-          raise InvalidResponse, "Sorry, I couldn't understand that number. Please try again.", caller
-        end
-        @value = BigDecimal.new(raw_input.to_s)
-      end
+      input = self.class.parse(raw_input)
+      number = self.class.valid?(input) && input
```

Comment:
Why bother assigning `number`? Given that `valid?` raises, couldn't you just do something like:

```ruby
input = self.class.parse(raw_input)
self.class.validate(input)
@value = BigDecimal.new(input.to_s)
```